### PR TITLE
remove some duplicated code

### PR DESCRIPTION
**Pull request by zehuaiWANG**

### CODE REVIEW

**Comment by zehuaiWANG** [comment]
@googlebot I signed it!

**Comment by zehuaiWANG** [comment]
@googlebot I signed it!

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -346,7 +347,7 @@ type startFunc func(stop <-chan struct{}) error
 // initMonitor initializes the configuration for the pilot monitoring server.
 func (s *Server) initMonitor(args *PilotArgs) error { //nolint: unparam
 	s.addStartFunc(func(stop <-chan struct{}) error {
-		monitor, addr, err := startMonitor(args.DiscoveryOptions.MonitoringAddr, s.mux)
+		monitor, addr, err := istiod.StartMonitor(args.DiscoveryOptions.MonitoringAddr, s.mux)
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/server.go]: pilot should not depend on istiod

**Comment by howardjohn** [comment]
@dougreid

File: pilot/pkg/bootstrap/server.go (hunk 1 above)
Comment (zehuaiWANG) [reply pilot/pkg/bootstrap/server.go]: So can we use these repeated codes as public libraries? In this way, the repetition of the whole file will bring difficulties to the  code reading later.

**Pull request closed without merging**